mikupad.html

```diff
@@ -1873,7 +2211,64 @@
 	return [value, updateState];
 }
 
+
+function serializeTemplateCollection(collection) {
+	const templatesArray = [...collection.templates.entries()].map(([id, {name, prefix, suffix}]) => ({id, name, prefix, suffix}));
+	return JSON.stringify({
+		templates: templatesArray,
+		nextId: collection.nextId,
+		selectedTemplateId: collection.selectedTemplateId,
+	});
+}
+
+function deserializeTemplateCollection(serialized) {
+	const parsed = JSON.parse(serialized);
+	const collection = new TemplateCollection();
+	collection.templates.clear();
+	collection.nextId = parsed.nextId;
+	collection.selectedTemplateId = parsed.selectedTemplateId;
+	parsed.templates.forEach(({id, name, prefix, suffix}) => {
+		collection.templates.set(id, {name, prefix, suffix});
+		collection.nextId = Math.max(collection.nextId, id + 1);
+	});
+	return collection;
+}
+
+function usePersistentTemplateCollection(name, initialState) {
+	const handleChangeRef = useRef(() => {});
+
+	const [templateStorage, setTemplateStorage] = useState(() => {
+		const init = () => {
+			try {
+				const item = localStorage.getItem(name);
+				return item ? deserializeTemplateCollection(item) : new TemplateCollection();
+			} catch (e) {
+				console.error(e);
+				return new TemplateCollection();
+			}
+		};
+		return init();
+	});
+
+	handleChangeRef.current = (newState) => {
+		setTemplateStorage(newState);
+		localStorage.setItem(name, serializeTemplateCollection(newState));
+	};
+
+	useEffect(() => {
+		if (templateStorage.setChangeListener && templateStorage.removeChangeListener) {
+			templateStorage.setChangeListener(handleChangeRef.current);
+			return () => {
+				templateStorage.removeChangeListener(handleChangeRef.current);
+			};
+		}
+	}, [templateStorage]);
+
+	return [templateStorage, setTemplateStorage];
+}
+
 export function App({ sessionStorage, useSessionState }) {
+	const [templateStorage, setTemplateStorage] = usePersistentTemplateCollection('templateStorage', new TemplateCollection());

```

Comment:
There's no need to add this in the App if it's only ever used by the Templates component, and as it is now, it will create a new instance of `TemplateCollection' each time the page is rendered.